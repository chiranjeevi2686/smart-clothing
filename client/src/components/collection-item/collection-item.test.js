import React from "react";
import { render, fireEvent } from '@testing-library/react';
import CollectionItem from './collection-item.component';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const defaultStore = {
    cart: {
        cartItems: [{
            id: 19,
            imageUrl: "https://i.ibb.co/mJS6vz0/blue-jean-jacket.png",
            name: "Blue Jean Jacket",
            price: 90,
            quantity: 1
        }],
        hidden: false,
        payment: null
    }
};
const mockStore = configureStore();
describe('Collection item component', () => {
    it('should render collection item', async () => {
        const itemProp = { imageUrl: 'www.test.com', price: '23', name: 'test' }
        const { asFragment, getByTestId } = render(<Provider store={mockStore(defaultStore)}>
            <CollectionItem item={itemProp} />
        </Provider>)
        expect(asFragment()).toMatchSnapshot();
        const addButton = getByTestId('add-cart-button');
        fireEvent.click(addButton);
    })
})
