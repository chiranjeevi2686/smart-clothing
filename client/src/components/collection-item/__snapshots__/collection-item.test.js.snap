// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection item component should render collection item 1`] = `
<DocumentFragment>
  <div
    class="sc-gsTCUz llOAWQ"
  >
    <div
      class="sc-hKgILt ifgiZx"
    />
    <div
      class="sc-eCssSg jFCvIw"
    >
      <span
        class="sc-jSgupP jbqeKv"
      >
        test
      </span>
      <span
        class="sc-gKsewC kIfwPX"
      >
        $23
      </span>
    </div>
    <button
      class="sc-bdfBwQ fByGxQ sc-dlfnbm iPOXsS"
      data-test-id="add-cart-button"
    >
      ADD TO CART
    </button>
  </div>
</DocumentFragment>
`;
